src/Beacon.jsx

```diff
@@ -99,14 +124,16 @@ export class Beacon extends Component {
       }
     }
     const targetElement = this.getTargetElement();
-    const targetBounds = targetElement.getBoundingClientRect();
-    const boundsChanged =
-      this.state.targetBounds.top !== targetBounds.top ||
-      this.state.targetBounds.bottom !== targetBounds.bottom ||
-      this.state.targetBounds.left !== targetBounds.left ||
-      this.state.targetBounds.right !== targetBounds.right;
-    if (boundsChanged || (this.state.targetElement !== targetElement)) {
-      this.setState({ targetElement, targetBounds }); // eslint-disable-line react/no-did-update-set-state
+    if (targetElement) {
+      const targetBounds = targetElement.getBoundingClientRect();
+      const boundsChanged =
+        this.state.targetBounds.top !== targetBounds.top ||
+        this.state.targetBounds.bottom !== targetBounds.bottom ||
+        this.state.targetBounds.left !== targetBounds.left ||
+        this.state.targetBounds.right !== targetBounds.right;
+      if (boundsChanged || (this.state.targetElement !== targetElement)) {
```

Comment:
Trying to minimize the number of `setState` calls since it seems like even if I call it twice in the same lifecycle method, it still forces two rerenders (not 100% sure about that though).